packages/test/test-utils/src/timeoutUtils.ts

```diff
@@ -25,9 +26,17 @@ export async function timeoutAwait<T = void>(
     return Promise.race([promise, timeoutPromise<T>(() => { }, timeoutOptions)]);
 }
 
-export async function ensureContainerConnected(container: Container): Promise<void> {
-    if (!container.connected) {
-        return timeoutPromise((resolve) => container.once("connected", () => resolve()));
+export async function ensureContainerConnected(
+    container: IContainer,
+    timeoutOptions: TimeoutWithError = {},
+    failOnContainerClose: boolean = true): Promise<void> {
+    if (container.connectionState !== ConnectionState.Connected) {
+        return timeoutPromise((resolve, reject) => {
```

Comment:
Just FYI, I have noticed that when I run the ODSP tests locally on codespaces a lot of tests time out because it doesn't get to connected state in 250ms (which is the default timeout in timeoutPromise). Not sure if that's an issue with my codespaces only but something to keep in mind if this happens in CI or something.

Reply:
I did get a bit concerned that it would be too short for some environments or particular tests, and was waiting to see if we started seeing frequent timeouts in CI. Maybe we make 1s the default specifically for `ensureContainerConnected`?

Reply:
I am not sure what a good value would be. Maybe start with 1 second and it can be adjusted based on test results? I would add a comment to the function either way saying that the timeout is x and it should be adjusted if tests timeout due to container taking more than x to connect.